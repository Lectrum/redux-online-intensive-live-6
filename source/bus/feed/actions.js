// Types
import { FILL_STARSHIPS } from './types';

export const fillStarships = (starships) => ({
    type:    FILL_STARSHIPS,
    payload: starships,
});
