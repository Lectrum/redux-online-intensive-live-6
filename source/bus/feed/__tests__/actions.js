// Actions
import { feedActions } from '../actions';

describe('feed actions', () => {
    test('startFetching', () => {
        expect(feedActions.startFetching()).toMatchSnapshot();
    });
    test('stopFetching', () => {
        expect(feedActions.stopFetching()).toMatchSnapshot();
    })
    test('fetchStarshipsAsync', () => {
        expect(feedActions.fetchStarshipsAsync()).toMatchSnapshot();
    })
    test('fillStarships', () => {
        expect(feedActions.fillStarships(__.feedState.starships)).toMatchSnapshot();
    })
});