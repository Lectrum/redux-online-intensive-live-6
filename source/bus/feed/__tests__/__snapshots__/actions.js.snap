// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed actions fetchStarshipsAsync 1`] = `
Object {
  "type": "FETCH_STARSHIPS_ASYNC",
}
`;

exports[`feed actions fillStarships 1`] = `
Object {
  "payload": Array [],
  "type": "FILL_STARSHIPS",
}
`;

exports[`feed actions startFetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`feed actions stopFetching 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
